diff --git a/crc32.cpp b/crc32.cpp
index a1ad3c5..303881b 100644
--- a/crc32.cpp
+++ b/crc32.cpp
@@ -5,12 +5,7 @@
 //
 
 #include "crc32.h"
-
-// big endian architectures need #define __BYTE_ORDER __BIG_ENDIAN
-#ifndef _MSC_VER
-#include <endian.h>
-#endif
-
+#include "portable_endian.h"
 
 /// same as reset()
 CRC32::CRC32()
diff --git a/keccak.cpp b/keccak.cpp
index cba1cae..6c74939 100644
--- a/keccak.cpp
+++ b/keccak.cpp
@@ -5,12 +5,7 @@
 //
 
 #include "keccak.h"
-
-// big endian architectures need #define __BYTE_ORDER __BIG_ENDIAN
-#ifndef _MSC_VER
-#include <endian.h>
-#endif
-
+#include "portable_endian.h"
 
 /// same as reset()
 Keccak::Keccak(Bits bits)
diff --git a/md5.cpp b/md5.cpp
index 2743319..657b8b2 100644
--- a/md5.cpp
+++ b/md5.cpp
@@ -5,11 +5,7 @@
 //
 
 #include "md5.h"
-
-#ifndef _MSC_VER
-#include <endian.h>
-#endif
-
+#include "portable_endian.h"
 
 /// same as reset()
 MD5::MD5()
diff --git a/portable_endian.h b/portable_endian.h
new file mode 100644
index 0000000..ad7bb85
--- /dev/null
+++ b/portable_endian.h
@@ -0,0 +1,391 @@
+// "License": Public Domain
+// I, Mathias Panzenböck, place this file hereby into the public domain. Use it at your own risk for whatever you like.
+// In case there are jurisdictions that don't support putting things in the public domain you can also consider it to
+// be "dual licensed" under the BSD, MIT and Apache licenses, if you want to. This code is trivial anyway. Consider it
+// an example on how to get the endian conversion functions on different platforms.
+
+#ifndef PORTABLE_ENDIAN_H__
+#define PORTABLE_ENDIAN_H__
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#if (defined(_WIN16) || defined(_WIN32) || defined(_WIN64)) && !defined(__WINDOWS__)
+
+#   define __WINDOWS__
+
+#endif
+
+#if defined(__linux__) || defined(__CYGWIN__)
+
+#   include <endian.h>
+
+#elif defined(__APPLE__)
+
+#   include <machine/endian.h>
+#   include <libkern/OSByteOrder.h>
+
+inline uint16_t htobe16(uint16_t x)
+{
+    return OSSwapHostToBigInt16(x);
+}
+
+inline uint16_t htole16(uint16_t x)
+{
+    return OSSwapHostToLittleInt16(x);
+}
+
+inline uint16_t be16toh(uint16_t x)
+{
+    return OSSwapBigToHostInt16(x);
+}
+
+inline uint16_t le16toh(uint16_t x)
+{
+    return OSSwapLittleToHostInt16(x);
+}
+
+inline uint32_t htobe32(uint32_t x)
+{
+    return OSSwapHostToBigInt32(x);
+}
+
+inline uint32_t htole32(uint32_t x)
+{
+    return OSSwapHostToLittleInt32(x);
+}
+
+inline uint32_t be32toh(uint32_t x)
+{
+    return OSSwapBigToHostInt32(x);
+}
+
+inline uint32_t le32toh(uint32_t x)
+{
+    return OSSwapLittleToHostInt32(x);
+}
+
+inline uint64_t htobe64(uint64_t x)
+{
+    return OSSwapHostToBigInt64(x);
+}
+
+inline uint64_t htole64(uint64_t x)
+{
+    return OSSwapHostToLittleInt64(x);
+}
+
+inline uint64_t be64toh(uint64_t x)
+{
+    return OSSwapBigToHostInt64(x);
+}
+
+inline uint64_t le64toh(uint64_t x)
+{
+    return OSSwapLittleToHostInt64(x);
+}
+
+#   define __BYTE_ORDER    BYTE_ORDER
+#   define __BIG_ENDIAN    BIG_ENDIAN
+#   define __LITTLE_ENDIAN LITTLE_ENDIAN
+#   define __PDP_ENDIAN    PDP_ENDIAN
+
+#elif defined(__OpenBSD__)
+
+#   include <endian.h>
+
+#   define __BYTE_ORDER    BYTE_ORDER
+#   define __BIG_ENDIAN    BIG_ENDIAN
+#   define __LITTLE_ENDIAN LITTLE_ENDIAN
+#   define __PDP_ENDIAN    PDP_ENDIAN
+
+#elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__)
+
+#   include <endian.h>
+
+#elif defined(__WINDOWS__)
+
+#  include <winsock2.h>
+#  if defined(_MSC_VER)
+#    include <netiodef.h>
+#    define LITTLE_ENDIAN  _LITTLE_ENDIAN
+#    define BIG_ENDIAN  _BIG_ENDIAN
+#    define PDP_ENDIAN _PDP_ENDIAN
+#  endif
+#  if defined(__GNUC__)
+#    include <sys/param.h>
+#  endif
+
+#  if (BYTE_ORDER == LITTLE_ENDIAN)
+
+inline uint16_t htobe16(uint16_t x)
+{
+    return htons(x);
+}
+
+inline uint16_t htole16(uint16_t x)
+{
+    return x;
+}
+
+inline uint16_t be16toh(uint16_t x)
+{
+    return ntohs(x);
+}
+
+inline uint16_t le16toh(uint16_t x)
+{
+    return x;
+}
+
+inline uint32_t htobe32(uint32_t x)
+{
+    return htonl(x);
+}
+
+inline uint32_t htole32(uint32_t x)
+{
+    return x;
+}
+
+inline uint32_t be32toh(uint32_t x)
+{
+    return ntohl(x);
+}
+
+inline uint32_t le32toh(uint32_t x)
+{
+    return x;
+}
+
+inline uint64_t htobe64(uint64_t x)
+{
+    return htonll(x);
+}
+
+inline uint64_t htole64(uint64_t x)
+{
+    return x;
+}
+
+inline uint64_t be64toh(uint64_t x)
+{
+    return ntohll(x);
+}
+
+inline uint64_t le64toh(uint64_t x)
+{
+    return x;
+}
+
+#  elif BYTE_ORDER == BIG_ENDIAN
+
+#    if defined(_MSC_VER)
+
+inline uint16_t __builtin_bswap16(uint16_t x)
+{
+    return _byteswap_ushort(x);
+}
+
+inline uint32_t __builtin_bswap32(uint32_t x)
+{
+    return _byteswap_ulong(x);
+}
+
+inline uint64_t __builtin_bswap64(uint64_t x)
+{
+    return _byteswap_uint64(x);
+}
+
+#    endif
+
+inline uint16_t htobe16(uint16_t x)
+{
+    return x;
+}
+
+inline uint16_t htole16(uint16_t x)
+{
+    return __builtin_bswap16(x);
+}
+
+inline uint16_t be16toh(uint16_t x)
+{
+    return x;
+}
+
+inline uint16_t le16toh(uint16_t x)
+{
+    return __builtin_bswap16(x);
+}
+
+inline uint32_t htobe32(uint32_t x)
+{
+    return x;
+}
+
+inline uint32_t htole32(uint32_t x)
+{
+    return __builtin_bswap32(x);
+}
+
+inline uint32_t be32toh(uint32_t x)
+{
+    return x;
+}
+
+inline uint32_t le32toh(uint32_t x)
+{
+    return __builtin_bswap32(x);
+}
+
+inline uint64_t htobe64(uint64_t x)
+{
+    return x;
+}
+
+inline uint64_t htole64(uint64_t x)
+{
+    return __builtin_bswap64(x);
+}
+
+inline uint64_t be64toh(uint64_t x)
+{
+    return x;
+}
+
+inline uint64_t le64toh(uint64_t x)
+{
+    return __builtin_bswap64(x);
+}
+
+#  else
+
+#    error byte order not supported
+
+#  endif
+
+#  define __BYTE_ORDER    BYTE_ORDER
+#  define __BIG_ENDIAN    BIG_ENDIAN
+#  define __LITTLE_ENDIAN LITTLE_ENDIAN
+#  define __PDP_ENDIAN    PDP_ENDIAN
+
+#elif defined(__QNXNTO__)
+
+#  include <gulliver.h>
+
+#  define __LITTLE_ENDIAN 1234
+#  define __BIG_ENDIAN    4321
+#  define __PDP_ENDIAN    3412
+
+#  if defined(__BIGENDIAN__)
+
+#    define __BYTE_ORDER __BIG_ENDIAN
+
+inline uint16_t htobe16(uint16_t x)
+{
+    return x;
+}
+
+inline uint32_t htobe32(uint32_t x)
+{
+    return x;
+}
+
+inline uint64_t htobe64(uint64_t x)
+{
+    return x;
+}
+
+inline uint16_t htole16(uint16_t x)
+{
+    return ENDIAN_SWAP16(x);
+}
+
+inline uint32_t htole32(uint32_t x)
+{
+    return ENDIAN_SWAP32(x);
+}
+
+inline uint64_t htole64(uint64_t x)
+{
+    return ENDIAN_SWAP64(x);
+}
+
+#  elif defined(__LITTLEENDIAN__)
+
+#    define __BYTE_ORDER __LITTLE_ENDIAN
+
+inline uint16_t htole16(uint16_t x)
+{
+    return x;
+}
+
+inline uint32_t htole32(uint32_t x)
+{
+    return x;
+}
+
+inline uint64_t htole64(uint64_t x)
+{
+    return x;
+}
+
+inline uint16_t htobe16(uint16_t x)
+{
+    return ENDIAN_SWAP16(x);
+}
+
+inline uint32_t htobe32(uint32_t x)
+{
+    return ENDIAN_SWAP32(x);
+}
+
+inline uint64_t htobe64(uint64_t x)
+{
+    return ENDIAN_SWAP64(x);
+}
+
+#  else
+
+#    error byte order not supported
+
+#  endif
+
+inline uint16_t be16toh(uint16_t x)
+{
+    return ENDIAN_BE16(x);
+}
+
+inline uint32_t be32toh(uint32_t x)
+{
+    return ENDIAN_BE32(x);
+}
+
+inline uint64_t be64toh(uint64_t x)
+{
+    return ENDIAN_BE64(x);
+}
+
+inline uint16_t le16toh(uint16_t x)
+{
+    return ENDIAN_LE16(x);
+}
+
+inline uint32_t le32toh(uint32_t x)
+{
+    return ENDIAN_LE32(x);
+}
+
+inline uint64_t le64toh(uint64_t x)
+{
+    return ENDIAN_LE64(x);
+}
+
+#else
+
+#   error platform not supported
+
+#endif
+
+#endif
diff --git a/sha1.cpp b/sha1.cpp
index 8331e92..197aa6b 100644
--- a/sha1.cpp
+++ b/sha1.cpp
@@ -5,12 +5,7 @@
 //
 
 #include "sha1.h"
-
-// big endian architectures need #define __BYTE_ORDER __BIG_ENDIAN
-#ifndef _MSC_VER
-#include <endian.h>
-#endif
-
+#include "portable_endian.h"
 
 /// same as reset()
 SHA1::SHA1()
diff --git a/sha256.cpp b/sha256.cpp
index 0c6138c..06743b1 100644
--- a/sha256.cpp
+++ b/sha256.cpp
@@ -5,12 +5,7 @@
 //
 
 #include "sha256.h"
-
-// big endian architectures need #define __BYTE_ORDER __BIG_ENDIAN
-#ifndef _MSC_VER
-#include <endian.h>
-#endif
-
+#include "portable_endian.h"
 
 /// same as reset()
 SHA256::SHA256()
diff --git a/sha3.cpp b/sha3.cpp
index 2f6bc7b..5fbea4c 100644
--- a/sha3.cpp
+++ b/sha3.cpp
@@ -5,11 +5,7 @@
 //
 
 #include "sha3.h"
-
-// big endian architectures need #define __BYTE_ORDER __BIG_ENDIAN
-#ifndef _MSC_VER
-#include <endian.h>
-#endif
+#include "portable_endian.h"
 
 #include <iostream>
 
