diff --git a/CMake/AbseilDll.cmake b/CMake/AbseilDll.cmake
index original..fixed 100644
--- a/CMake/AbseilDll.cmake
+++ b/CMake/AbseilDll.cmake
@@ -457,7 +457,7 @@ list(APPEND ABSL_INTERNAL_DLL_FILES
   "strings/string_view.h"
 )
 
-if(MSVC)
+if(WIN32)
   list(APPEND ABSL_INTERNAL_DLL_FILES
     "time/internal/cctz/src/time_zone_name_win.cc"
     "time/internal/cctz/src/time_zone_name_win.h"
