diff --git a/include/TGUI/StringView.hpp b/include/TGUI/StringView.hpp
index 48180ab..34ee69a 100644
--- a/include/TGUI/StringView.hpp
+++ b/include/TGUI/StringView.hpp
@@ -28,11 +28,9 @@
 #if !TGUI_EXPERIMENTAL_USE_STD_MODULE
     #include <string>
     #include <cctype> // tolower
-
+    #include <algorithm>
     #if TGUI_COMPILED_WITH_CPP_VER >= 17
         #include <string_view>
-    #else
-        #include <algorithm> // min
     #endif
 #endif
 
