diff --git a/source/gui/dragdrop.cpp b/source/gui/dragdrop.cpp
index 5711fbd..6b732d2 100644
--- a/source/gui/dragdrop.cpp
+++ b/source/gui/dragdrop.cpp
@@ -25,7 +25,7 @@
 #	include <windows.h>
 #	include <oleidl.h>
 #	include <comdef.h>
-#	include <Shlobj.h>
+#	include <shlobj.h>
 #elif defined(NANA_X11)
 #	include "../detail/posix/xdnd_protocol.hpp"
 #	include <nana/gui/detail/native_window_interface.hpp>
diff --git a/source/gui/filebox.cpp b/source/gui/filebox.cpp
index 63586ab..0b26f9f 100644
--- a/source/gui/filebox.cpp
+++ b/source/gui/filebox.cpp
@@ -22,7 +22,7 @@
 #	ifndef NANA_MINGW	//<Shobjidl.h> isn't supported well on MinGW
 #		include <Shobjidl.h>
 #	else
-#		include <Shlobj.h>
+#		include <shlobj.h>
 #	endif
 #elif defined(NANA_POSIX)
 #	include <nana/gui/widgets/label.hpp>
